### ISSUE TITLE
Volume sharing performance issues

### ISSUE THREAD

**Issue by jmreicha** [boot2docker/boot2docker#593]
As of the 1.3 release, volume mounting seems to be working.  The issue is that when attempting to share a large git repo (17k files) in to the container on boot2docker I am getting some crazy slow performance.  Are there any workarounds or plans to address this issue?

The problem seems to point at VirtualBox Guest Additions when you have lots of small files - http://mitchellh.com/comparing-filesystem-performance-in-virtual-machines

Maybe NFS shared folders could be used somehow?  Or maybe VirtualBox people are working on fixing the performance issues?

**Comment by leighmcculloch** [comment]
I attempted to use volume mounting to sync a local web project with ~12k small files and it caused the web app to take from 4 seconds to 1 minute to respond to a single request. Host volumes are unusable for development on a decent sized project.

I'm now using `fswatch` and `unison` to do fast automatic two-way sync. I've wrapped this up in a volume container: https://github.com/leighmcculloch/docker-unison to make it plug-and-play.

**Comment by md5** [comment]
@leighmcculloch you might want to have a look at https://github.com/gansbrest/hodor

**Comment by lox** [comment]
Interesting idea @leighmcculloch, the only bummer is that it eliminates the nice characteristics of having `/Users` mounted in the docker host.

**Comment by leighmcculloch** [comment]
@lox agreed. I've actually switched to using docker inside Vagrant with NFS. I started seeing problems with the shared volume link between the two containers disconnecting. Vagrant with NFS is a simpler setup, but the syncing with NFS is not as fast as unison was.

**Comment by brikis98** [comment]
For folks still struggling with the performance of mounted folders and vboxsf, check out [A productive development environment with Docker on OS X](http://www.ybrikman.com/writing/2015/05/19/docker-osx-dev/).

**Comment by STRML** [comment]
Thank you! I've been using NFS, but this is very promising as an
alternative!

**Comment by thaJeztah** [comment]
@brikis98 Excellent article! You should submit it for the Docker Weekly newsletter (sorry, don't know where to do that, but I can try and find out if you need)

**Comment by brikis98** [comment]
@thaJeztah: thanks! From a quick search, it looks like you need to tweet at the Docker Twitter account to get into the newsletter. I'll try that, but if anyone has other suggestions, send them my way.

**Comment by thaJeztah** [comment]
@brikis98 that could well be the way to do that. If I hear another way, I'll let you know :+1:

**Comment by mhworth** [comment]
@brikis98, Thanks! docker-osx-dev works great.

**Comment by fyddaben** [comment]
i have the   problem too, is there any easy way to slove it ?? i want to use  it in my team

**Comment by pencilcheck** [comment]
Still wish for a native solution directly from virtualbox or boot2docker

**Comment by jordscream** [comment]
No temporary Stable solution for Windows ?

**Comment by fyddaben** [comment]
@jordscream  u can try sshd

**Comment by jordscream** [comment]
@fyddaben Can you be more specific ? :/

Today The fact is:
1/ docker machine runs almost good in windows 7 except the shared vboxsf of Vbox which is very slow. Change vboxsf to NFS, CIFS, SMB is a complicated stuff and generate severals errors (symlinks, difficult setup for designers in Windows...)
2/ Boot2docker Vagrant Box is an alternative to the problem but the setup takes several hours if everything is okay... The final is something which work but with big problems as "cannot enable tty mode"... Crazy errors if you have a docker image displaying something...

The need is to have a docker in windows with shared volumes for developpers (web designers in my case) by following the normal process of docker install (docker toolbox) and have a "correct" shared volume performance...

If you have something easier than 1/ and faster/better than 2/ , I Take :P

And Yeah, I think everybody sould go to Linux or Mac but in big companies, my bad, Windows stays the King...

**Comment by fyddaben** [comment]
@jordscream    u can install a sshd client on windows. and  this is the [website](https://osxfuse.github.io/)
```
sshfs -p 2023 -o allow_other work@$(docker-machine ip dev):/docker/path/  /widows/path
``` 
2023 is my ssh open port. 


and   the speed should be   faster than vbox

**Comment by jordscream** [comment]
@fyddaben it seems your website supports only OSX, alternative for windows ?
Thanks for your help
